### PR TITLE
Add docs for unstable_dataStrategy and unstable_skipActionErrorRevalidation

### PR DESCRIPTION
**Pull request by brophdawg11**

### CODE REVIEW

**Comment by brophdawg11** [comment]
Todo: add a warning to the top of data strategy docs explaining the risks

File: docs/routers/create-browser-router.md [hunk 1]
```diff
@@ -181,6 +182,179 @@ const router = createBrowserRouter(
...
+}
+```
+
+`unstable_dataStrategy` receives the same arguments as a `loader`/`action` (`request`, `params`) but it also receives a `matches` array which is an array of the matched routes where each match is extended with 2 new fields for use in the data strategy function:
+
+- `match.resolve` - An async function that will resolve any `route.lazy` implementations and execute the route's handler (if necessary), returning a `HandlerResult`
+  - You should call `match.resolve` for _all_ matches every time to ensure that all lazy routes are properly resolved
+  - This does not mean you're calling the loader/action (the "handler") - resolve will only call the handler internally if needed and if you don['t pass your own `handlerOverride` function parameter
+  - See the examples below for how to implement custom handler execution via `match.resolve`
+- `match.shouldLoad` - A boolean value indicating whether this route handler needs go be called in this pass
```
Comment (kiliman) [inline docs/routers/create-browser-router.md:235]:
```diff
-needs go be called
+needs to be called
```

File: docs/routers/create-browser-router.md [hunk 2]
```diff
@@ -181,6 +182,179 @@ const router = createBrowserRouter(
...
+In the simplest case, let's look at hooking into this API to add some logging for when our route loaders/actions execute:
+
+```ts
+let router = createBrowserRouter(routes, {
+  unstable_dataStrategy({ request, matches }) {
+    return Promise.all(
+      matches.map(async (match) => {
+        console.log(`Processing route ${match.route.id}`);
+        // Don't override anything - just resolve route.lazy + call loader
+        let result = await m.resolve();
```
Comment (kiliman) [inline docs/routers/create-browser-router.md:258]:
```diff
-let result = await m.resolve()
+let result = await match.resolve()
```

File: docs/routers/create-browser-router.md [hunk 3]
```diff
@@ -181,6 +182,179 @@ const router = createBrowserRouter(
...
+  result: any; // data, Error, Response, DeferredData
+  status?: number;
+}
+```
+
+`unstable_dataStrategy` receives the same arguments as a `loader`/`action` (`request`, `params`) but it also receives a `matches` array which is an array of the matched routes where each match is extended with 2 new fields for use in the data strategy function:
+
+- `match.resolve` - An async function that will resolve any `route.lazy` implementations and execute the route's handler (if necessary), returning a `HandlerResult`
+  - You should call `match.resolve` for _all_ matches every time to ensure that all lazy routes are properly resolved
+  - This does not mean you're calling the loader/action (the "handler") - resolve will only call the handler internally if needed and if you don['t pass your own `handlerOverride` function parameter
```
Comment (kiliman) [inline docs/routers/create-browser-router.md:233]:
```diff
-if you don['t pass
+if you don't pass
```

**Review by kiliman — COMMENTED** [review]
Sorry didn't mean to hit review, just leaving comments.

**Pull request merged**